src/lib/lspace/geom/affinexform2.rs

```diff
@@ -0,0 +1,443 @@
+    }
+}
+
+// MULTIPLY BY Point2; apply scale and translation
+
+impl Mul<Point2> for AffineXform2 {
+    type Output = Point2;
+
+    fn mul(self, p: Point2) -> Point2 {
+        return Point2::new(p.x * self.v[0][0] + p.y * self.v[0][1] + self.v[0][2],
+                           p.x * self.v[1][0] + p.y * self.v[1][1] + self.v[1][2]);
+    }
+}
+
+impl <'a> Mul<Point2> for &'a AffineXform2 {
+    type Output = Point2;
+
+    fn mul(self, p: Point2) -> Point2 {
+        return Point2::new(p.x * self.v[0][0] + p.y * self.v[0][1] + self.v[0][2],
+                           p.x * self.v[1][0] + p.y * self.v[1][1] + self.v[1][2]);
+    }
+}
+
+impl <'a> Mul<&'a Point2> for AffineXform2 {
+    type Output = Point2;
+
+    fn mul(self, p: &'a Point2) -> Point2 {
+        return Point2::new(p.x * self.v[0][0] + p.y * self.v[0][1] + self.v[0][2],
+                           p.x * self.v[1][0] + p.y * self.v[1][1] + self.v[1][2]);
+    }
+}
+
+impl <'a, 'b> Mul<&'b Point2> for &'a AffineXform2 {
+    type Output = Point2;
+
+    fn mul(self, p: &'b Point2) -> Point2 {
+        return Point2::new(p.x * self.v[0][0] + p.y * self.v[0][1] + self.v[0][2],
+                           p.x * self.v[1][0] + p.y * self.v[1][1] + self.v[1][2]);
+    }
+}
+
+// MULTIPLY BY AffineXform2; matrix multiplication
+
+impl Mul<AffineXform2> for AffineXform2 {
+    type Output = AffineXform2;
+
+    fn mul(self, x: AffineXform2) -> AffineXform2 {
+        return AffineXform2{v: [
+                [
+                    self.v[0][0] * x.v[0][0] + self.v[0][1] * x.v[1][0],
+                    self.v[0][0] * x.v[0][1] + self.v[0][1] * x.v[1][1],
+                    self.v[0][0] * x.v[0][2] + self.v[0][1] * x.v[1][2] + self.v[0][2],
+                ],
+                [
+                    self.v[1][0] * x.v[0][0] + self.v[1][1] * x.v[1][0],
+                    self.v[1][0] * x.v[0][1] + self.v[1][1] * x.v[1][1],
+                    self.v[1][0] * x.v[0][2] + self.v[1][1] * x.v[1][2] + self.v[1][2],
+                ],
+            ]};
+    }
+}
+
+impl <'a> Mul<AffineXform2> for &'a AffineXform2 {
+    type Output = AffineXform2;
+
+    fn mul(self, x: AffineXform2) -> AffineXform2 {
+        return AffineXform2{v: [
+                [
+                    self.v[0][0] * x.v[0][0] + self.v[0][1] * x.v[1][0],
+                    self.v[0][0] * x.v[0][1] + self.v[0][1] * x.v[1][1],
+                    self.v[0][0] * x.v[0][2] + self.v[0][1] * x.v[1][2] + self.v[0][2],
+                ],
+                [
+                    self.v[1][0] * x.v[0][0] + self.v[1][1] * x.v[1][0],
+                    self.v[1][0] * x.v[0][1] + self.v[1][1] * x.v[1][1],
+                    self.v[1][0] * x.v[0][2] + self.v[1][1] * x.v[1][2] + self.v[1][2],
+                ],
+            ]};
+    }
+}
+
+impl <'a> Mul<&'a AffineXform2> for AffineXform2 {
+    type Output = AffineXform2;
+
+    fn mul(self, x: &'a AffineXform2) -> AffineXform2 {
+        return AffineXform2{v: [
```

Comment:
Is it just me or could this block be indented one level to the left? [I think there are other similar blocks below too.]